src/utils.ts

```diff
@@ -292,3 +300,65 @@ export const processNewOpenClosed = (
 
   return updatedNewOpenClosed;
 };
+
+export const getLatestTxHash = async (
+  programId: PublicKey,
+  connection: Connection
+) => {
+  const [signature] = await connection.getSignaturesForAddress(
+    programId,
+    { limit: 1 },
+    "confirmed"
+  );
+  return signature.signature;
+};
+
+export const fetchPoolsWithTicks = async (
+  retries: number = 0,
```

Comment:
tbh it doesn't make sense to make first argument optional. It would be better to accept retires as last parameter if it is optional, but in our case it will be always defined